File: models/webPushSubscriptionSqlModel.ts
import {
  DataTypes,
  Model,
  ModelStatic
} from 'sequelize'
import sequelize from '@/config/sequelize'
import WebPushSubscriptionSqlRecord from '@/types/WebPushSubscriptionSqlRecord'
import createId from '@/utilities/createId'
const webPushSubscriptionSqlModel: ModelStatic<Model<WebPushSubscriptionSqlRecord>> = sequelize.models.WebPushSubscription ?? sequelize.define<Model<WebPushSubscriptionSqlRecord>>(
  'WebPushSubscription', {
    ...createId(),
    access_token_id: {
      type: DataTypes.UUID,
      allowNull: true,
      references: {
        model: 'OAuthAccessToken',
        key: 'id'
      }
    },
    user_id: {
      type: DataTypes.UUID,
      allowNull: false,
      references: {
        model: 'User',
        key: 'id'
      }
    },
    endpoint: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    key_p256dh: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    key_auth: {
      type: DataTypes.TEXT,
      allowNull: false
    },
    data: {
      type: DataTypes.TEXT,
      allowNull: true
    }
  }, {
    tableName: 'web_push_subscriptions',
    timestamps: true
  }
)
export default webPushSubscriptionSqlModel

File: nomad/models/web_push_subscription_sql_model.py
"""
Web push SQL table model
"""
from uuid import uuid4
from django.db import models
class WebPushSubscription(models.Model):
    """
    Web push subscription SQL table model
    """
    class Meta:
        """
        Web push subscription SQL table model meta class
        """
        verbose_name = 'Web Push Subscription'
        verbose_name_plural = f'{verbose_name}s'
        db_table = verbose_name_plural.lower().replace(
            ' ',
            '_'
        )
    id = models.UUIDField(
        primary_key = True,
        default = uuid4,
        editable = False
    )
    access_token_id = models.ForeignKey(
        to = 'knox.AuthToken',
        on_delete = models.DO_NOTHING,
        related_name = 'web_push_subscriptions',
        null = True,
        blank = True
    )
    user_id = models.ForeignKey(
        to = 'auth.User',
        on_delete = models.CASCADE,
        related_name = 'web_push_subscriptions'
    )
    endpoint = models.TextField()
    key_p256dh = models.TextField()
    key_auth = models.TextField()
    data = models.TextField(
        null = True,
        blank = True
    )
    created_at = models.DateTimeField(
        auto_now_add = True
    )
    updated_at = models.DateTimeField(
        auto_now = True
    )
